//Define the used version of proto
syntax = "proto3";

package isr.simulation;

//Define a message to hold the features input by the client
message InitRequest {
    string StartDate      = 1 ;
    string EndDate        = 2 ;
}
//Define a message to hold the predicted price
message InitResponse {
    bool isRunning      = 1 ;
}

message StepRequest {
  int32 numSteps = 1;
}

message StepResponse {
  bool isDone = 1;
}

message EmissionsRequest { }

message EmissionsResponse {
  map<string, float> emissions = 1;
}

message ChangeCellStateRequest {
  map<string, bool> cell_state = 1;
}

message ChangeCellStateResponse {  }



//Define the service
service Simulate {
  rpc start_simulation(InitRequest) returns (InitResponse);
  rpc step(StepRequest) returns (StepResponse);
  rpc get_emissions(EmissionsRequest) returns(EmissionsResponse);
  rpc change_cell_state(ChangeCellStateRequest) returns(ChangeCellStateResponse);
}
